src/login/AuthProviderBase.ts

```diff
@@ -31,32 +35,32 @@ export abstract class AuthProviderBase {
 		context.subscriptions.push(this.outputChannel);
 	}
 
-	public abstract loginWithoutLocalServer(clientId: string, environment: Environment, isAdfs: boolean, tenantId: string): Promise<AbstractLoginResult>;
-	public abstract loginWithAuthCode(code: string, redirectUrl: string, clientId: string, environment: Environment, tenantId: string): Promise<AbstractLoginResult>;
-	public abstract loginWithDeviceCode(environment: Environment, tenantId: string): Promise<AbstractLoginResult>;
-	public abstract loginSilent(environment: Environment, storedCreds: string, tenantId: string): Promise<AbstractLoginResult>;
+	public abstract loginWithoutLocalServer(clientId: string, environment: Environment, isAdfs: boolean, tenantId: string): Promise<TLoginResult>;
+	public abstract loginWithAuthCode(code: string, redirectUrl: string, clientId: string, environment: Environment, tenantId: string): Promise<TLoginResult>;
+	public abstract loginWithDeviceCode(environment: Environment, tenantId: string): Promise<TLoginResult>;
+	public abstract loginSilent(environment: Environment, tenantId: string, migrateToken?: boolean): Promise<TLoginResult>;
 	public abstract getCredentials(environment: string, userId: string, tenantId: string): AbstractCredentials;
-	public abstract getCredentials2(environment: Environment, userId: string, tenantId: string): AbstractCredentials2;
-	public abstract updateSessions(environment: Environment, loginResult: AbstractLoginResult, sessions: AzureSession[]): Promise<void>;
-	public abstract clearTokenCache(): Promise<void>;
-	public abstract deleteRefreshTokens(): Promise<void>;
+	public abstract getCredentials2(environment: Environment, userId: string, tenantId: string, accountInfo?: AccountInfo): AbstractCredentials2;
+	public abstract updateSessions(environment: Environment, loginResult: TLoginResult, sessions: AzureSession[]): Promise<void>;
+	public abstract clearLibraryTokenCache(): Promise<void>;
```

Comment:
I don't think we need both `clearLibraryTokenCache` and `clearLocalTokenCache`. Just keep it all in one place (`clearTokenCache`) like it was before